### PR TITLE
add clean to makefile

### PR DESCRIPTION
**Pull request by mefuller**
`make clean` now available to wipe out items created locally during testing, `__pycache__` folders; `PHONY` added to protect arguments

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -12,3 +12,9 @@ gcn:
 
 gcn-cpu:
 	bash devtools/install_gcn_cpu.sh
+
+clean:
+	find -type d -name __pycache__ -exec rm -rf {} +
+	rm -rf testing
```
Comment (alongd) [inline Makefile:20]: I'm not certain the test folders should be here. ARC should delete al testing folders after it's done testing. It is possible that if a test fails, then the testing folder persists, but is this what we're addressing?

**Review by alongd — COMMENTED** [review]
Looking good. I added one question.
Could you rebase so the tests will pass?

File: Makefile (hunk 1 above)
Comment (mefuller) [reply Makefile]: Yes

**Comment by mefuller** [comment]
I broke something during rebase - see #454

**Pull request closed without merging**